p2p/host/eventbus/basic_metrics.go

```diff
@@ -0,0 +1,139 @@
+package eventbus
+
+import (
+	"reflect"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const metricNamespace = "libp2p_eventbus"
+
+var (
+	eventsEmitted = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: metricNamespace,
+			Name:      "events_emitted_total",
+			Help:      "Events Emitted",
+		},
+		[]string{"event"},
+	)
+	notificationTime = prometheus.NewHistogramVec(
+		prometheus.HistogramOpts{
+			Namespace: metricNamespace,
+			Name:      "notification_time_microseconds",
+			Help:      "Time taken to notify all subscribers",
+			Buckets:   prometheus.ExponentialBuckets(1.0, 2, 10),
+		},
+		[]string{"event"},
+	)
+	numSubscribers = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Namespace: metricNamespace,
+			Name:      "num_subscribers",
```

Comment:
```suggestion
			Name:      "subscribers_total",
```

Reply:
In Grafana, this would look pretty cool as a Gauge: 
<img width="1365" alt="image" src="https://user-images.githubusercontent.com/1478487/216519035-4fa60f3f-ea66-4d5d-8829-1eded5ce735b.png">

Reply:
fixed